frontend/app/vault/index/route.js

```diff
@@ -6,5 +6,15 @@ export default Route.extend({
 
   model() {
     return this.get('vault').list()
+  },
+
+  actions: {
+    willTransition(transition) {
+      if (transition.targetName === 'vault.index.index') {
+        this.set('controller.detail', false)
+      } else if (transition.targetName === 'vault.index.edit') {
+        this.set('controller.detail', true)
+      }
+    }
```

Comment:
Make `detail` a computed property on the controller.. You can use the router services `currentRouteName`